packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts

```diff
@@ -0,0 +1,72 @@
+import { ComponentRequest, ComponentUpdateRequest } from '@aws-sdk/client-iottwinmaker';
+
+import { IAnchorComponent, KnownComponentType } from '../../interfaces';
+import { componentTypeToId } from '../../common/entityModelConstants';
+
+import { createDataBindingMap } from './dataBindingUtils';
+
+export const createTagEntityComponent = (tag: IAnchorComponent): ComponentRequest => {
+  const comp: ComponentRequest = {
+    componentTypeId: componentTypeToId[KnownComponentType.Tag],
+    properties: {},
+  };
+  if (tag.icon) {
+    comp.properties!['icon'] = {
```

Comment:
I guess these exclamation marks can be removed?

Reply:
the type of `properties ` in `ComponentRequest` can be undefined and typescript is not smart enough to know that I've assigned value to it above

Reply:
Maybe we can do this check (comp.properties && comp.properties[icon])?